test/server/config_validation/xds_verifier.cc

```diff
@@ -81,10 +81,37 @@ void XdsVerifier::listenerUpdated(const envoy::config::listener::v3::Listener& l
   ENVOY_LOG_MISC(debug, "About to update listener {} to {}", listener.name(), getRoute(listener));
   dumpState();
 
-  if (std::any_of(listeners_.begin(), listeners_.end(), [&](auto& rep) {
+  const auto existing_warming =
+      std::find_if(listeners_.begin(), listeners_.end(), [&](const auto& rep) {
         return rep.listener.name() == listener.name() &&
-               getRoute(listener) == getRoute(rep.listener) && rep.state != DRAINING;
-      })) {
+               getRoute(rep.listener) == getRoute(listener) && rep.state == WARMING;
+      });
+  const auto existing_active =
+      std::find_if(listeners_.begin(), listeners_.end(), [&](const auto& rep) {
+        return rep.listener.name() == listener.name() &&
+               getRoute(rep.listener) == getRoute(listener) && rep.state == ACTIVE;
+      });
+
+  // return early if the listener is a duplicate
```

Comment:
Capitalize the first letter before you are rejected by Matt :)

Reply:
All of the comments in `xds_verifier.cc` have a small first letter, I'll go through and change them all